flink-runtime/src/main/java/org/apache/flink/runtime/io/network/buffer/LocalBufferPool.java

```diff
@@ -396,7 +396,7 @@ private MemorySegment requestMemorySegment(int targetChannel) {
         synchronized (availableMemorySegments) {
             checkDestroyed();
 
-            if (availableMemorySegments.isEmpty()) {
+            if (availableMemorySegments.isEmpty() && isRequestedSizeReached()) {
```

Comment:
I'm not sure whether `availableMemorySegments.poll();` should be executed when `availableMemorySegments.isEmpty()`. 

1. How about call it just when `!availableMemorySegments.isEmpty()`?
2. We should add some comments for `requestOverdraftMemorySegmentFromGlobal`, explain when to request the overdraft buffer.

```suggestion
            if (!availableMemorySegments.isEmpty()) {
                segment = availableMemorySegments.poll();
            } else if (isRequestedSizeReached()) {
                // Only when the buffer request reaches the upper limit, requests an overdraft buffer
                segment = requestOverdraftMemorySegmentFromGlobal();
            }
```

Reply:
IIUC, poll buffer from empty `availableMemorySegments ` is not necessary as we guarded this field by lock.